src/viam/examples/mlmodel/example_audio_classification_client.cpp

```diff
@@ -221,9 +231,14 @@ int main(int argc, char* argv[]) try {
         // secret. Please see other examples for more details on
         // connecting to robots with the C++ SDK.
         vsdk::DialOptions dial_opts;
-        dial_opts.set_credentials(vsdk::Credentials(opt_robot_secret.get()));
-        auto robot =
-            vsdk::RobotClient::at_address(opt_robot_host.get(), {0, {std::move(dial_opts)}});
+        dial_opts.set_type("api-key");
```

Comment:
I think you can keep the condensed form of this rather than unpacking it into named variables so much. Something like:
```
vsdk::DialOptions dial_opts;
dial_opts.set_type("api-key");
dial_opts.set_entity(opt_api_key.get());
dial_opts.set_credentials(Credentials{opt_api_key_id.get()});
auto robot = vsdk::RobotClient::at_address(opt_robot_host.get(), {0, {std::move(dial_opts)}});
```

You might not even need that `Credentials{...}`. If not then
```
dial_opts.set_credentials({opt_api_key_id.get()});
```

Reply:
Will try

Reply:
@npentrel - Did it work?

Reply:
I got it to compile now but not with the code that is staged in the app... I haven't run it, since I don't have anything confnigured to try this but might try tomorrow